src/assets/scripts/airport/StandardRoute/RouteSegmentModel.js

```diff
@@ -1,30 +1,79 @@
+import _forEach from 'lodash/forEach';
 import _map from 'lodash/map';
+import _isArray from 'lodash/isArray';
 import _uniqId from 'lodash/uniqueId';
 import StandardRouteWaypointModel from './StandardRouteWaypointModel';
 
 /**
+ * Provides an interface for dealing with a list of `StandardRouteWaypointModel`s that make up a given route segment.
+ *
  * @class RouteSegmentModel
  */
 export default class RouteSegmentModel {
     /**
+     * segmentWaypoints should come in a similar shape to:
+     * - ["_NAPSE068", "NAPSE", ["RIOOS", "A130+"], "COMPS"]
+     *
      * @constructor
-     * @param name {string}
-     * @param segmentWaypoints {array}
+     * @param name {string}  Icao of particular waypoint
```

Comment:
This usage of `ICAO` doesn't really make sense. It's been (mis)used throughout the application in various places for a while actually. [ICAO](http://www.icao.int/Pages/default.aspx) is the the United Nations of airplane stuff, and they maintain their ICAO Document 8643, which contains unique and globally accepted identifiers for aircraft types, and Document 7910 for airport identifiers, etc. So it makes sense in that context, but it has been kind of globally taken to be synonymous with `identifier` or `name`, which it is not. Not that it really matters, but just thought I'd share.